### ISSUE TITLE
bug: cloudfront s3 origin requiring bucket name in path

### ISSUE THREAD

**Issue by vt-rc** [localstack/localstack#10431]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

First I create a distribution with s3 origin

```
    response = client.create_distribution(
        DistributionConfig={
            "CallerReference": str(datetime.now()),
            "Aliases": {
                "Quantity": 1,
                "Items": [
                    f"{bucket_name}.aws.localdev",
                ],
            },
            "Origins": {
                "Quantity": 1,
                "Items": [
                    {
                        "Id": "s3-origin",
                        "DomainName": f"{bucket_name}.s3.localhost.localstack.cloud:4566",
                        "S3OriginConfig": {
                            "OriginAccessIdentity": "",
                        },
                    },
                ],
            },
            "DefaultCacheBehavior": {
                "TargetOriginId": "s3-origin",
                "ViewerProtocolPolicy": "allow-all",
                "AllowedMethods": {
                    "Quantity": 2,
                    "Items": [
                        "HEAD",
                        "GET",
                    ],
                    "CachedMethods": {
                        "Quantity": 2,
                        "Items": [
                            "HEAD",
                            "GET",
                        ],
                    },
                },
                "SmoothStreaming": False,
                "DefaultTTL": 86400,
                "MinTTL": 0,
                "MaxTTL": 86400 * 2,
                "Compress": False,
                "LambdaFunctionAssociations": {
                    "Quantity": 0,
                },
                "FieldLevelEncryptionId": "",
            },
            "Comment": f"cdn for {bucket_name}",
            "Enabled": True,
        }
    )
```

then I upload a file to s3 to using filezilla pro. I then try to access the file using a browser. I have tried this with cloudfront domain and custom domain by pointing both domain to 127.0.0.1 in /etc/hosts. The url is like {custom_domain}/{s3_filekey} or {distribution_domain}/{s3_filekey}.  both time the url path returns

```
<Error>
<Code>NoSuchBucket</Code>
<Message>The specified bucket does not exist</Message>
<RequestId>959b04d3-9574-4bad-9f1a-f9af586de697</RequestId>
<BucketName>github-recovery-codes.txt</BucketName>
</Error>
```

because it expects bucket name at the first folder in the path. 

### Expected Behavior

On AWS, I believe when cloudfront access an S3 origin, it won't need bucket name prepended to the filekey. 

if I access the same filekey using {bucket_name}.s3.localhost.localstack.cloud:4566/{filekey}, it can access fine

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- OS:
- LocalStack:
```

**Comment by elbazon** [comment]
+1

**Comment by rfenner** [comment]
I can report that this is still happening as I just ran into it.

My S3 setup looks like this
`resource "aws_s3_bucket" "cc_video" {
  bucket = local.s3_cdn_name
}

resource "aws_s3_bucket_ownership_controls" "s3_owner" {
  bucket = aws_s3_bucket.cc_video.id
  rule {
    object_ownership = "BucketOwnerEnforced"
  }
}

resource "aws_s3_bucket_versioning" "s3_versioning" {
  bucket = aws_s3_bucket.cc_video.id
  versioning_configuration {
    status = "Disabled"
  }
}

resource "aws_s3_bucket_public_access_block" "s3_bucket_access" {
  bucket                  = aws_s3_bucket.cc_video.id
  block_public_acls       = true
  block_public_policy     = true
  ignore_public_acls      = true
  restrict_public_buckets = true
}
`

And my cloudfront looks like this
`resource "aws_cloudfront_origin_access_control" "site_oac" {
  name                              = "site"
  origin_access_control_origin_type = "s3"
  signing_behavior                  = "always"
  signing_protocol                  = "sigv4"
}

resource "aws_cloudfront_origin_access_control" "cdn_oac" {
  name                              = "cdn"
  origin_access_control_origin_type = "s3"
  signing_behavior                  = "always"
  signing_protocol                  = "sigv4"
}

resource "aws_cloudfront_distribution" "cc_cdn_site" {
  enabled = true

  origin {
    origin_access_control_id = aws_cloudfront_origin_access_control.site_oac.id
    domain_name = aws_s3_bucket.cc_site.bucket_regional_domain_name
    origin_id   = local.site_origin_id
    custom_origin_config {
      http_port              = 80
      https_port             = 443
      origin_protocol_policy = "https-only"
      origin_ssl_protocols = ["SSLv3" , "TLSv1" , "TLSv1.1" , "TLSv1.2"]
    }
  }

  origin {
    origin_access_control_id = aws_cloudfront_origin_access_control.cdn_oac.id
    domain_name = aws_s3_bucket.cc_video.bucket_regional_domain_name
    origin_id   = local.cdn_origin_id
    custom_origin_config {
      http_port              = 80
      https_port             = 443
      origin_protocol_policy = "https-only"
      origin_ssl_protocols = ["SSLv3" , "TLSv1" , "TLSv1.1" , "TLSv1.2"]
    }
  }

  default_cache_behavior {
    allowed_methods = ["GET","HEAD", "OPTIONS"]
    cached_methods = ["GET","HEAD"]
    target_origin_id       = local.site_origin_id

    forwarded_values {
      query_string = true

      cookies {
        forward = "all"
      }
    }

    viewer_protocol_policy = "redirect-to-https"
    min_ttl = 0
    default_ttl = 150
    max_ttl = 300
  }

  restrictions {
    geo_restriction {
      restriction_type = "none"
    }
  }

  viewer_certificate {
    acm_certificate_arn = local.acm_arn
  }
  price_class = "PriceClass_200"

 `

I'm using the stable pro version published around a month ago.

I can confirm if i add the bucket name after the / before the key path it downloads the file.
`https://cdn.local.crash-clips.com:8080/my-cdn-bucket/published/79cd1536-22db-11f0-9bfb-cac7490f8525/test-video.mp4`
but this doesn't
`https://cdn.my-custom-domain.com:8080/published/79cd1536-22db-11f0-9bfb-cac7490f8525/test-video.mp4`
